spec/models/manageiq/providers/amazon/cloud_manager/refresh_worker_spec.rb

```diff
@@ -1,13 +1,8 @@
 describe ManageIQ::Providers::Amazon::CloudManager::RefreshWorker do
   context "EMS with children" do
-    let!(:network_manager) { FactoryBot.create(:ems_network) }
-    let!(:storage_manager) { FactoryBot.create(:ems_storage) }
-    let(:ems) do
-      FactoryBot.create(:ems_cloud).tap do |ems|
-        network_manager.update_attributes(:parent_ems_id => ems.id)
-        storage_manager.update_attributes(:parent_ems_id => ems.id)
-      end
-    end
+    let!(:ems) { FactoryBot.create(:ext_management_system) }
```

Comment:
This should probably be `ems_amazon` and `ems_amazon_network`, right?

Reply:
I mean I'd say yes if this wasn't going to fail for a name validation error now? Which makes no sense to me but if I had to guess I'd have to ask you if [the change we made yesterday](https://github.com/ManageIQ/manageiq/commit/730a7492b8aa5df83a26dcfef94aaabfa152efc8) needs something else related to naming?

Reply:
Also these are a freaking pain in the nevermind.

Reply:
uh.... WAT?

![image](https://user-images.githubusercontent.com/651659/60343142-4fca4e00-9981-11e9-807a-3951d726be58.png)

Reply:
:sob: I told you, it's just disgusting. I can't even. Can you even? I don't understand.
:laughing:

Reply:
Oh wierd... It looks like creating an `ems_amazon` creates all 3, so you'll just have to do the lookup to set the other 2 `let`s
```ruby
irb(main):015:0> p= FactoryBot.create(:ems_amazon)
=> #<ManageIQ::Providers::Amazon::CloudManager id: 34000000000796, name: "ems_0000000000010", created_on: "2019-06-28 12:49:52", updated_on: "2019-06-28 12:49:52", guid: "c6f559af-cf89-4598-a03c-a7a238a142a6", zone_id: 34000000001619, type: "ManageIQ::Providers::Amazon::CloudManager", api_version: nil, uid_ems: nil, host_default_vnc_port_start: nil, host_default_vnc_port_end: nil, provider_region: "us-east-1", last_refresh_error: nil, last_refresh_date: nil, provider_id: nil, realm: nil, tenant_id: nil, project: nil, parent_ems_id: nil, subscription: nil, last_metrics_error: nil, last_metrics_update_date: nil, last_metrics_success_date: nil, tenant_mapping_enabled: nil, enabled: true, options: nil, zone_before_pause_id: nil, last_inventory_date: nil>
irb(main):016:0> ExtManagementSystem.count
=> 3
irb(main):017:0> ExtManagementSystem.pluck(:name)
=> ["ems_0000000000010", "ems_0000000000010 Network Manager", "ems_0000000000010 EBS Storage Manager"]
```

Reply:
thanks, i hate it.

Reply:
the second test in this file makes no valid sense then, since it's gonna make a leaf class that has three too.

Reply:
also do we need to let people know somehow that we've made this change?

Reply:
A blog post may be a good idea.